src/blocks/keyboard_layout.rs

```diff
@@ -318,3 +330,93 @@ trait LocaleBusInterface {
     #[dbus_proxy(property, name = "X11Variant")]
     fn variant(&self) -> zbus::Result<String>;
 }
+
+#[dbus_proxy(
+    interface = "ru.gentoo.kbdd",
+    default_service = "ru.gentoo.KbddService",
+    default_path = "/ru/gentoo/KbddService"
+)]
+trait KbddBusInterface {
+    #[dbus_proxy(signal, name = "layoutNameChanged")]
+    fn layout_updated(&self, layout: String) -> zbus::Result<()>;
+
+    #[dbus_proxy(name = "getCurrentLayout")]
+    fn current_layout_index(&self) -> zbus::Result<u32>;
+
+    #[dbus_proxy(name = "getLayoutName")]
+    fn current_layout(&self, layout_id: u32) -> zbus::Result<String>;
+}
+
+struct KbddBus {
+    stream: layoutNameChangedStream<'static>,
+    layout: String,
+    variant: Option<String>,
```

Comment:
Since you are already storing both layout and variant, you can pack them in one filed of type `Info`